### PR TITLE
support multiple devices

### PR DESCRIPTION
**Pull request by toanju**
to support different link types cnetlink is refactored to simplify
addition of new link types like vxlan

### CODE REVIEW

File: src/netlink/cnetlink.cpp [hunk 1]
```diff
@@ -23,12 +23,24 @@
...
       bridge(nullptr), nl_proc_max(10), running(false), rfd_scheduled(false),
-      l3(swi, tap_man, this) {
+      l3(swi, tap_man, this), lt2names{lt_names} {
+
+  assert(lt2names.size() == LT_MAX);
+
+  enum link_type lt = LT_UNKNOWN;
+  for (const auto &n : lt2names) {
+    assert(lt != LT_MAX);
+    LOG(INFO) << n;
```
Comment (DFritzsche) [inline src/netlink/cnetlink.cpp]: a wild log message

File: src/netlink/cnetlink.cpp [hunk 2]
```diff
@@ -23,12 +23,24 @@
 #define ROUTE_CAST(obj) reinterpret_cast<struct rtnl_route *>(obj)
 #define ADDR_CAST(obj) reinterpret_cast<struct rtnl_addr *>(obj)
 
+#define lt_names "unknown", "bridge", "tun"
```
Comment (DFritzsche) [inline src/netlink/cnetlink.cpp]: why no integrate this into `enum link_type`?

Comment (toanju) [reply src/netlink/cnetlink.cpp]: these are the name that besides `unknow` match the result of rtnl_link_get_type (i.e. the identifier/kind of the netlink operations).

File: src/netlink/cnetlink.cpp (hunk 1 above)
Comment (toanju) [reply src/netlink/cnetlink.cpp]: wild thing... yup that has to go away

**Pull request merged**